### PR TITLE
Remove unneeded steps from lint CI workflow

### PR DESCRIPTION
**Pull request by twz123**
## Description

The build environment is never used in the workflow, and the Golang steps are built inside Docker.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by mikhail-sakhnov** [comment]
is it actually work without build? the codegen part is used to be done as a part of the build step and some linters fail without generated code.

**Comment by twz123** [comment]
> is it actually work without build? the codegen part is used to be done as a part of the build step and some linters fail without generated code.

The codegen part is actually built (w/o embedded binaries). But that happens from within Docker, so no need to have the Go toolchain be installed on the GH runner.

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -38,14 +38,6 @@ jobs:
...
-        run: .github/workflows/prepare-build-env.sh
-
-      - name: Set up Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: ${{ env.GO_VERSION }}
-
       - name: Go caches
         uses: actions/cache@v3
         with:
```
Comment (kke) [inline .github/workflows/lint.yaml:43]: I guess it doesn't use the cache then either, at least not like that. There's some kind of docker layer caching thing for github runners.

Comment (twz123) [reply .github/workflows/lint.yaml:43]: The cache should be used. It will be mounted into the container.

Comment (kke) [reply .github/workflows/lint.yaml:43]: Right

**Review by mikhail-sakhnov — APPROVED** [review]

**Pull request merged**